.circleci/src/workflows/e2e_tests.yml

```diff
@@ -0,0 +1,45 @@
+when:
+  or:
+    - and:
+        - << pipeline.parameters.api_triggered >>
+        - equal: [<< pipeline.parameters.workflow_type >>, e2e_tests]
+    - and:
+        - equal: [main, << pipeline.git.branch >>]
+        - not: << pipeline.parameters.api_triggered >>
```

Comment:
Just to clarify because I'm confused by CircleCI's syntax, this means: run if someone triggers the e2e tests via API or if we're on the branch main and it hasn't been triggered by the API?

If that's right, then I'm not sure why we wouldn't want to run the tests on main for a manual trigger?

Reply:
It would run on main for a manual trigger, as the first condition (first half of the or) is true then.